File: frontend/src/types/navigation.ts
export type NavigationAction = {
  id: number;
  action_key: string;
  action_title: string;
  permission_key: string;
  button_color: string | null;
  button_icon: string | null;
  sort_order: number;
};

export type NavigationMenu = {
  id: number;
  parent_menu_id: number | null;
  menu_key: string;
  menu_title: string;
  route_path: string | null;
  icon: string | null;
  permission_key: string | null;
  sort_order: number;
  menu_level: number;
  is_expandable: boolean;
  actions: NavigationAction[];
  children: NavigationMenu[];
};

export type NavigationGroup = {
  id: number;
  group_key: string;
  group_title: string;
  group_icon: string | null;
  group_badge: string | null;
  group_color: string | null;
  sort_order: number;
  menus: NavigationMenu[];
};

File: backend/app/schemas/navigation.py
from __future__ import annotations

from pydantic import BaseModel, Field


class NavigationActionResponse(BaseModel):
    id: int
    action_key: str
    action_title: str
    permission_key: str
    button_color: str | None = None
    button_icon: str | None = None
    sort_order: int


class NavigationMenuResponse(BaseModel):
    id: int
    parent_menu_id: int | None = None
    menu_key: str
    menu_title: str
    route_path: str | None = None
    icon: str | None = None
    permission_key: str | None = None
    sort_order: int
    menu_level: int = 1
    is_expandable: bool = False
    actions: list[NavigationActionResponse] = Field(default_factory=list)
    children: list["NavigationMenuResponse"] = Field(default_factory=list)


class NavigationGroupResponse(BaseModel):
    id: int
    group_key: str
    group_title: str
    group_icon: str | None = None
    group_badge: str | None = None
    group_color: str | None = None
    sort_order: int
    menus: list[NavigationMenuResponse] = Field(default_factory=list)


NavigationMenuResponse.model_rebuild()
